mlflow/server/handlers.py

```diff
@@ -1352,6 +1378,9 @@ def _validate_source(source: str, run_id: str) -> None:
             INVALID_PARAMETER_VALUE,
         )
 
+    if is_http_uri(source) or is_mlflow_artifacts_uri(source) or is_local_uri(source):
+        _validate_non_local_source_contains_relative_paths(source)
```

Comment:
```suggestion
    _validate_non_local_source_contains_relative_paths(source)
```

Reply:
Great idea :) Updated, changed the Exception message, and added a few more conditions in the test suite based on our discussion (ftp, s3, etc)